draft-ietf-tls-tls13.md

```diff
@@ -1854,14 +1820,6 @@ subsequent connections.
 The "key_share" extension contains the endpoint's cryptographic parameters
 for non-PSK key establishment methods (currently DHE or ECDHE).
 
-Clients which offer one or more (EC)DHE cipher suites MUST send this
-extension and SHOULD send at least one supported KeyShareEntry value.
-Servers MUST NOT negotiate any of these cipher suites unless a supported
-value was provided.
-If this extension is not provided in a ServerHello or ClientHello,
```

Comment:
I think it's still missing client behavior around:

* Server signaled it wanted to do ECDHE but we didn't get a key_share.
* Server signaled it didn't want to do ECDHE but we did get a key_share.

(This is the cost of doing the more redundant PSK-based server indication rather than using ServerHello extensions.)